packages/babel-traverse/src/scope/index.js

```diff
@@ -90,7 +90,9 @@ const collectorVisitor = {
     //if (path.isFlow()) return;
 
     // we've ran into a declaration!
-    path.scope.getFunctionParent().registerDeclaration(path);
+    const parent =
+      path.scope.getFunctionParent() || path.scope.getProgramParent();
```

Comment:
This wouldn't work, because `getFunctionParent()` either returns the function path or throws. It never returns a falsy value and thus `getProgramParent()` is never called.